src/gui/Measurement.cc

```diff
@@ -50,11 +50,43 @@ void Measurement::startMeasureAngle(void)
 }
 QString Measurement::statemachine(QPoint mouse) 
 {
-  if(qglview->measure_state == MEASURE_IDLE) return "";
+  if(qglview->measure_state == MEASURE_IDLE) return {};
   qglview->selectPoint(mouse.x(),mouse.y());
-  double ang=NAN;
   double dist=NAN;
   SelectedObject obj1, obj2, obj3;
+  SelectedObject ruler = {
+	  .type=SelectionType::SELECTION_INVALID
+  };
+  auto display_angle = [this] (Vector3d p1, Vector3d p2, Vector3d p3, Vector3d p4) 
+  {	
+	SelectedObject ruler;
+	Vector3d side1, side2;
```

Comment:
These could be declared where they're assigned. That way they can also be const.